lib/writer.rb

```diff
@@ -0,0 +1,32 @@
+require 'json'
+
+class Writer
+
+  def initialize(path, file = nil)
+    @path = path
+    @file = file
+  end
+
+  def run(contacts)
+    file.nil? ? write(contacts) : tests_write(contacts)
+  end
+
+  private
+
+  attr_reader :path, :file
+
+  def parse(contacts)
+    contacts.to_json
+  end
+
+  def write(contacts)
+    File.open(path, "w+") do |file|
```

Comment:
Thanks Jim! :sparkles: 

You are totally right about the local file variable, I should have renamed it. It was a bit clearer that I wanted to use the attr_reader when I didn't have the block.

Regarding `flush` and `close`, you mean that I can flush the data instead of closing the file everytime I want to actually write it to the file?
We didn't try `flush` but all our attempts of writing were in vain unless we closed it.
Gonna try flush out and see what happens.